### PR TITLE
MBS-9395: Fix pagination on userstats table

### PR DESCRIPTION
**Pull request by TobiGa**

### CODE REVIEW

**Comment by TobiGa** [comment]
> I think also the upper set_count_sql in line 101 is broken? I think this should also be fixed.

I changed it to resemble the select query.

File: classes/local/userstats_table.php [hunk 1]
```diff
@@ -91,26 +91,28 @@ public function __construct(
...
+                    . 'LEFT JOIN {local_ai_manager_userinfo} ui ON rl.userid = ui.userid '
+                    . 'JOIN {user} u ON u.id = rl.userid';
+            $where = $tenantfield . ' = :tenant AND purpose = :purpose GROUP BY u.id ORDER BY lastname';
             $params = [
                     'tenant' => $tenant->get_sql_identifier(),
                     'purpose' => $purpose,
             ];
             $this->set_count_sql(
-                    "SELECT COUNT(DISTINCT userid) FROM {local_ai_manager_request_log} rl JOIN {user} u ON rl.userid = u.id "
+                    "SELECT COUNT(DISTINCT rl.userid) FROM {local_ai_manager_request_log} rl "
```
Comment (PhMemmel) [inline classes/local/userstats_table.php:103]: Maybe also use the variable $from here? Would make the code clearer.

File: classes/local/userstats_table.php [hunk 2]
```diff
@@ -91,26 +91,28 @@ public function __construct(
         $tenantfield = get_config('local_ai_manager', 'tenantcolumn');
         if (!empty($purpose)) {
             $fields = 'u.id as id, lastname, firstname, locked, COUNT(value) AS requestcount, SUM(value) AS currentusage';
-            $from = '{local_ai_manager_request_log} rl LEFT JOIN {local_ai_manager_userinfo} ui ON rl.userid = ui.userid'
-                    . ' JOIN {user} u ON u.id = rl.userid';
-            $where = $tenantfield . ' = :tenant AND purpose = :purpose GROUP BY u.id';
+            $from = '{local_ai_manager_request_log} rl '
+                    . 'LEFT JOIN {local_ai_manager_userinfo} ui ON rl.userid = ui.userid '
+                    . 'JOIN {user} u ON u.id = rl.userid';
+            $where = $tenantfield . ' = :tenant AND purpose = :purpose GROUP BY u.id ORDER BY lastname';
```
Comment (PhMemmel) [inline classes/local/userstats_table.php:97]:
```suggestion
            $where = $tenantfield . ' = :tenant AND purpose = :purpose GROUP BY u.id ORDER BY u.lastname';
```

Not really necessary of course, but is more uniform

File: classes/local/userstats_table.php [hunk 3]
```diff
@@ -91,26 +91,28 @@ public function __construct(
...
                     . "WHERE " . $tenantfield . " = :tenant AND purpose = :purpose",
                     $params
             );
         } else {
             $fields = 'u.id as id, lastname, firstname, COUNT(value) AS requestcount';
-            $from =
-                    '{local_ai_manager_request_log} rl LEFT JOIN {user} u ON rl.userid = u.id';
-            $where = 'u.' . $tenantfield . ' = :tenant GROUP BY u.id';
+            $from = '{local_ai_manager_request_log} rl LEFT JOIN {user} u ON rl.userid = u.id';
+            $where = 'u.' . $tenantfield . ' = :tenant GROUP BY u.id ORDER BY lastname';
```
Comment (PhMemmel) [inline classes/local/userstats_table.php:112]:
```suggestion
            $where = 'u.' . $tenantfield . ' = :tenant GROUP BY u.id ORDER BY u.lastname';
```

File: purpose_statistics.php [hunk 4]
```diff
@@ -65,7 +65,7 @@
     echo html_writer::div(get_string('userwithusageonlyshown', 'local_ai_manager'));
 
     $table = new \local_ai_manager\local\userstats_table($uniqid, $purpose, $tenant, $baseurl);
-    $table->out(5, false);
+    $table->out(20, false);
```
Comment (PhMemmel) [inline purpose_statistics.php:68]: Perfect, 20 is a reasonable value I guess :+1:

**Review by PhMemmel — CHANGES_REQUESTED** [review]

File: classes/local/userstats_table.php (hunk 1 above)
Comment (TobiGa) [reply classes/local/userstats_table.php:103]: Yep.

**Review by PhMemmel — APPROVED** [review]

**Pull request merged**